### ISSUE TITLE
keepFilesOnDelete not removing all files

### ISSUE THREAD

**Issue by kojot1234** [FriendsOfCake/cakephp-upload#465]
Problem only occurs when the first file in the field list is empty but the next one exists. 

Since `afterDelete` runs through all configuration fields if it fails on the first field it exits and will not remove any other files that were marked for removal with `keepFilesOnDelete`.

` afterDelete` should either continue on unsuccessful deletion or delete should not happen at all.

UploadBehavior.php
```
public function afterDelete(Event $event, Entity $entity, ArrayObject $options)
{
     ...
     if ((new Collection($success))->contains(false)) {
          continue; // return false;
     }
     ...
}
```

**Comment by saeideng** [comment]
you see `deleteCallback` option ?

**Comment by kojot1234** [comment]
Yes, that's how I got around right now but it seems like an unneeded workaround. I'm assuming the intended functionality should loop through and remove all fields set in the configuration and not quit on the first removal. Correct me if I'm wrong.

**Comment by josegonzalez** [comment]
Mind making a pull request?
